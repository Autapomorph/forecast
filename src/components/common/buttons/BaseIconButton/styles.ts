import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledIconButton = styled.span`
  cursor: pointer;

  :hover {
    color: var(--light-gray);
  }
`;
