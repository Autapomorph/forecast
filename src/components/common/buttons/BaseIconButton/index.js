import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { StyledIconButton } from './styles';

const BaseIconButton = ({ icon, size, onClick = () => {}, ...props }) => (
  <StyledIconButton {...props}>
    <FontAwesomeIcon icon={icon} size={size} onClick={onClick} />
  </StyledIconButton>
);

export default BaseIconButton;
