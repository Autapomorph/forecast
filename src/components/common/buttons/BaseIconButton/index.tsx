import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconProp, SizeProp } from '@fortawesome/fontawesome-svg-core';

import * as S from './styles';

type Props = {
  icon: IconProp;
  size?: SizeProp;
  onClick?: (event: React.MouseEvent<SVGSVGElement, MouseEvent>) => void;
};

const BaseIconButton = ({ icon, size = 'lg', onClick, ...props }: Props): React.ReactElement => (
  <S.IconButton {...props}>
    <FontAwesomeIcon icon={icon} size={size} onClick={onClick} />
  </S.IconButton>
);

export default BaseIconButton;
