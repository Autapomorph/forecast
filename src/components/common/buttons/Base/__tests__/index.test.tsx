import React from 'react';
import { shallow } from 'enzyme';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import BaseIconButton from '..';

describe('BaseIconButton tests', () => {
  it('renders properly', () => {
    // @ts-ignore
    const wrapper = shallow(<BaseIconButton icon="ufo" />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render FontAwesomeIcon', () => {
    const wrapper = shallow(<BaseIconButton icon="ufo" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
    expect(wrapper.find(FontAwesomeIcon).prop('icon')).toBe('ufo');
  });

  it('should handle clicks', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<BaseIconButton icon="ufo" onClick={onClick} />);
    wrapper.simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
