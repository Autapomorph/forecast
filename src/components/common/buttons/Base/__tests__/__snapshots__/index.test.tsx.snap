// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseIconButton tests renders properly 1`] = `
<styles__IconButton
  onClick={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="ufo"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="lg"
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</styles__IconButton>
`;

exports[`BaseIconButton tests should render FontAwesomeIcon 1`] = `
<styles__IconButton
  onClick={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="ufo"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="lg"
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</styles__IconButton>
`;
